repo/repo.go

```diff
@@ -98,16 +98,46 @@ func ListRepositories(workspace *rule.File) (repos []*rule.Rule, repoFileMap map
 			if err != nil {
 				return nil, nil, err
 			}
+			leveled := strings.HasPrefix(f, "+")
+			f = strings.TrimPrefix(f, "+")
 			f = filepath.Join(filepath.Dir(workspace.Path), filepath.Clean(f))
 			macroFile, err := rule.LoadMacroFile(f, "", defName)
 			if err != nil {
 				return nil, nil, err
 			}
 			for _, repo := range macroFile.Rules {
-				if name := repo.Name(); name != "" {
-					repos = append(repos, repo)
-					repoFileMap[name] = macroFile
-					repoIndexMap[name] = len(repos) - 1
+				if kind := repo.Kind(); kind == "go_repository" || !leveled {
```

Comment:
Using the OR not leveled - I avoid needing any additional logic

Reply:
Maybe:
```go
if repo.Kind() ==  "go_repository" {
//...
} else if leveled {
//...
}

Reply:
We can still support other kinds of repos when `leveled`:
```go
if name := repo.Name(); name != "" {
//...
} else leveled {
// treat all function calls without `name` as macros
}
```